src/domain/prompt/AdvancedSelectOnePrompt.ts

```diff
@@ -17,13 +17,19 @@
  * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  **/
 
-import {BasePrompt, IAdvancedSelectOne, IAdvancedSelectOnePromptConfig, ValidationException} from '../..'
+import {assertNotNull, BasePrompt, IAdvancedSelectOne, IAdvancedSelectOnePromptConfig, ValidationException} from '../..'
 
 export const ADVANCED_SELECT_ONE_PROMPT_KEY = 'AdvancedSelectOne'
 
 export class AdvancedSelectOnePrompt extends BasePrompt<IAdvancedSelectOnePromptConfig> {
-  validate(selectedRow?: IAdvancedSelectOne[]): boolean {
-    const {choiceRows, choiceRowFields, isResponseRequired} = this.config
+  validate(selectedRow?: IAdvancedSelectOne[], choiceRows?: string[][]): boolean {
```

Comment:
do we pass in the `choiceRows` value anywhere?
all `validate()` calls seem to have only one argument...

Reply:
It would be special for this. It's because FlowRunner can't directly access a CSV, that's up to the App to do. In our easiest/lowest complexity method of implementing this, we are using a 2 dimmensional string array, so we are just giving that to flowrunner directly. We can cycle back and implement something more complex if this ends up not being fast enough, but it's not worth doing that until it's necessary to do so. It's the problem of "Pre-mature Optimization" (You lose more time optimizing something too early as it may not have needed optimization, or it could change in the near future, leading to wasted effort)

Reply:
So what will happen when the flowrunner calls `validate()` with one argument in `isInputRequiredFor()`? will it fail?

What I mean is - how do we use this validate function? I thought we didn't use it directly in the clipboard app, but the flowrunner did it for us. In that case, doesn't it need to be changed too?